src/clarity-angular/data/datagrid/providers/selection.spec.ts

```diff
@@ -57,18 +57,31 @@ export default function(): void {
             expect(selectionInstance.current).toEqual([4, 2]);
         });
 
-        it("can select/deselect all items at once in multi selection type", function() {
+        it("can select/deselect all items at once in multi selection type if no pagination exist", function() {
             selectionInstance.selectionType = SelectionType.Multi;
             selectionInstance.toggleAll();
             expect(selectionInstance.current).toEqual(itemsInstance.displayed);
             selectionInstance.toggleAll();
             expect(selectionInstance.current).toEqual([]);
+
             selectionInstance.setSelected(4, true);
             expect(selectionInstance.current).toEqual([4]);
             selectionInstance.toggleAll();
             expect(selectionInstance.current.sort(numberSort)).toEqual(itemsInstance.displayed);
         });
 
+        it("can select/deselect items only on the current page", function() {
+            selectionInstance.selectionType = SelectionType.Multi;
+            pageInstance.size = 3;
+            pageInstance.current = 1;
+            selectionInstance.current = [1, 2];
```

Comment:
Can you make the current selection `[4, 1, 2]`? That can happen if the user first selects an item on page 2, then goes back to page 1.